src/submission/validation-clinical/sample.ts

```diff
@@ -1,6 +1,91 @@
-import { SubmissionValidationError, DataValidationErrors } from '../submission-entities';
+import {
+  SubmittedClinicalRecord,
+  FieldsEnum,
+  DataValidationErrors,
+  ValidatorResult,
+  ModificationType,
+} from '../submission-entities';
+import { DeepReadonly } from 'deep-freeze';
+import { Donor, Sample, Specimen } from '../../clinical/clinical-entities';
+import { FileType } from '../submission-api';
+import * as utils from './utils';
 
-export const validate = async (records: any): Promise<SubmissionValidationError[]> => {
+export const validate = async (
+  newDonorRecords: DeepReadonly<{ [clinicalType: string]: SubmittedClinicalRecord }>,
+  existentDonor: DeepReadonly<Donor>,
+): Promise<ValidatorResult> => {
   // this is a dummy error
-  return [];
+  const sampleRecord = newDonorRecords[FileType.SAMPLE];
+
+  // Preconditions
+  if (!utils.checkDonorRegistered(existentDonor, sampleRecord)) {
+    return {
+      type: ModificationType.ERRORSFOUND,
+      index: sampleRecord.index,
+      resultArray: [
+        utils.buildSubmissionError(
+          sampleRecord,
+          DataValidationErrors.ID_NOT_REGISTERED,
+          FieldsEnum.submitter_donor_id,
+        ),
+      ],
+    };
+  }
+  const specimen = utils.getRegisteredSubEntityInCollection(
+    FieldsEnum.submitter_specimen_id,
+    sampleRecord,
+    existentDonor.specimens,
+  ) as Specimen;
+  if (!specimen) {
+    return {
+      type: ModificationType.ERRORSFOUND,
```

Comment:
We're building this object a lot... should there be some code to do this in a consistent way?

Reply:
some method like you already have the `utils.buildSubmissionError`...